src/main/java/io/vertx/core/http/impl/headers/HeadersMultiMap.java

```diff
@@ -332,7 +332,16 @@ public String get(String name) {
   public void forEach(Consumer<? super Map.Entry<String, String>> action) {
     HeadersMultiMap.MapEntry e = head.after;
     while (e != head) {
-      action.accept(new AbstractMap.SimpleEntry<>(e.key.toString(), e.value.toString()));
+      action.accept(e.stringEntry());
+      e = e.after;
+    }
+  }
+
+  public void forEach(BiConsumer<String, String> action) {
+    HeadersMultiMap.MapEntry e = head.after;
+    while (e != head) {
+      Map.Entry<String, String> consumedEntry = e.stringEntry();
```

Comment:
in this case actually we don't need to call stringEntry, we can directly call `getkey()` / `getValue()` and then `toString()`, which avoids to create an entry when it contains a non string char sequence

Reply:
Okay, updated